src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java

```diff
@@ -0,0 +1,177 @@
+                    provider.setDefaultHeader("Accept", "application/json");
+                    provider.setDefaultHeader("Authorization", "Basic "
+                            + buildLoginData(username, password));
+                    result = provider.post("");
+                    addAccount(new String(result.getBody()));
+                    callback.onSuccess(new String(result.getBody()));
+                } catch (MalformedURLException e) {
+                    Timber.d(SecretStoreAuthzModule.class.getSimpleName(), "Error with URL", e);
+                    callback.onFailure(e);
+                } catch (HttpException e) {
+                    Timber.d(SecretStoreAuthzModule.class.getSimpleName(), "HTTP error", e);
+                    callback.onFailure(e);
+                } catch (RuntimeException e) {
+                    Timber.d(SecretStoreAuthzModule.class.getSimpleName(), "Connection Exception", e);
+                    callback.onFailure(e);
+                } catch (JSONException e) {
+                    Timber.d(SecretStoreAuthzModule.class.getSimpleName(), "Json parse Exception", e);
+                    callback.onFailure(e);
+                }
+            }
+        }).start();
+    }
+
+    private String buildLoginData(String username, String password) {
+        String cred = username + ":" + password;
+        return new String(Base64.encode(cred.getBytes(), Base64.NO_WRAP));
+    }
+
+    @Override
+    public boolean refreshAccess() {
+        return true;
+    }
+
+    @Override
+    public void deleteAccount() {
+        sessionStore.remove(AuthData.NAME);
+    }
+
+    @Override
+    public AuthorizationFields getAuthorizationFields(URI requestUri, String method, byte[] requestBody) {
+        AuthorizationFields fields = new AuthorizationFields();
+        OAuth2AuthzSession storedAccount = sessionStore.read(AuthData.NAME);
+        String cred = new String(Base64.encode((storedAccount.getAccessToken()).getBytes(), Base64.NO_WRAP));
+        fields.addHeader("Authorization", "Basic " + cred);
+        return fields;
+    }
+
+    @Override
+    public ModuleFields loadModule(URI relativeURI, String httpMethod, byte[] requestBody) {
+        AuthorizationFields authzFields = getAuthorizationFields(relativeURI, httpMethod, requestBody);
+        ModuleFields moduleFields = new ModuleFields();
+        moduleFields.setHeaders(authzFields.getHeaders());
+        moduleFields.setQueryParameters(authzFields.getQueryParameters());
+        return moduleFields;
+    }
+
+    @Override
+    public boolean handleError(HttpException exception) {
+        return false;
+    }
+
+    private void addAccount(String data) throws JSONException {
+        JSONObject auth = new JSONObject(data);
+        OAuth2AuthzSession session = new OAuth2AuthzSession();
+        session.setAccountId(AuthData.NAME);
+        session.setAccessToken(auth.getString("key") + ":" + auth.getString("secret"));
```

Comment:
@jpkrohling  I was just using that to store. But right, Its not a good practice. repairing it